### PR TITLE
Add assume_safety_prior_to_version config option

### LINKED ISSUE #12: Easy way to ignore unsafe operations in pre-existing migrations
My codebase has 164 migrations already, many of which include "unsafe" operations (likely created before the app was even in production).

Is there an easy way to "assure safety" for all migrations before a certain version number?

### PR DESCRIPTION
**Pull request by vergenzt**
Fixes #12 and #3.

Let me know what you think!

### CODE REVIEW

**Comment by vergenzt** [comment]
Adding changelog and readme entries right now, will amend.

**Comment by ankane** [comment]
Hey @vergenzt, thanks for the PR 👍  It looks like it's causing the tests for fail. Also, left a few comments inline.

File: test/strong_migrations_test.rb [hunk 1]
```diff
@@ -207,6 +221,18 @@ def test_create_table_force
...
+  def test_old_remove_column_assumed_safe
+    StrongMigrations.assume_safety_prior_to_version = SAFETY_ASSUMED_VERSION_CUTOFF
+    assert_safe OldRemoveColumnAssumedSafe
+    StrongMigrations.assume_safety_prior_to_version = 0
+  end
+
+  def test_new_remove_column_not_assumed_safe
+    StrongMigrations.assume_safety_prior_to_version = SAFETY_ASSUMED_VERSION_CUTOFF
+    assert_unsafe NewRemoveColumnNotAssumedSafe
+    StrongMigrations.assume_safety_prior_to_version = 0
```
Comment (ankane) [inline test/strong_migrations_test.rb]: Let's use `ensure` here.

File: CHANGELOG.md [hunk 2]
```diff
@@ -32,3 +33,5 @@
 ## 0.1.0
 
 - First release
+
+[@vergenzt]: https://github.com/vergenzt
```
Comment (ankane) [inline CHANGELOG.md]: The current changelog doesn't list contributors. Let's keep it consistent.

Comment (vergenzt) [inline CHANGELOG.md]: Sorry about that. Last repo I contributed to did this and I didn't think about it.

**Comment by vergenzt** [comment]
Thanks for the review @ankane! As I was looking at the tests I realized that success or failure of the tests depended on execution order, so I refactored the tests a bit in 2b005e6 to wipe the database and create the users table before each test case so nothing bleeds through. I believe I've addressed all comments.

Lmk what you think!

**Comment by ankane** [comment]
Thanks for addressing. There should be a way to make tests work without wiping the DB (as they do now). I'd prefer to keep the changeset to a minimum.

**Comment by vvondra** [comment]
This would be much appreciated!

**Comment by ankane** [comment]
Hey @vergenzt, I cleaned this up and merged in 50f9827d5bc580dae5ec8c4c24bd822f283e0ee3. Thanks for driving this. I think it'll be a very useful feature. 👍

File: lib/strong_migrations/migration.rb [hunk 3]
```diff
@@ -69,6 +69,14 @@ def method_missing(method, *args, &block)
 
     private
 
+    def safety_overridden?
+      @safe || ENV["SAFETY_ASSURED"] || should_assume_safety_due_to_version?
+    end
+
+    def should_assume_safety_due_to_version?
+      version && version <= StrongMigrations.assume_safety_prior_to_version
```
Comment (nisanth074) [inline lib/strong_migrations/migration.rb]: @vergenzt @ankane In what cases is `version` nil?

I just added the strong_migrations gem to our Rails 3 app. Unfortunately, the `start_after` option doesn't seem to work. Old dangerous migrations fail despite me configuring `start_after` option https://gist.github.com/nisanth074/ea45c8cdd50d4a08b53cb97b91fc3a6b

I added a `binding.pry` statement above this line and ran `bundle exec rake db:create db:migrate` again. It seems like `version` is always `nil` for all our migrations. So, `should_assume_safety_due_to_version?` returns nil and strong_migrations raises an error.

Can you let me know in what cases is `version` nil? Is Rails 3 supported?

Thanks for your time.

**Comment by ijones16** [comment]
Thanks for your work on this feature. It helped me out with a 5 year old project 🙌

**Comment by jpwhitaker** [comment]
I just found this thread after installing [rpush](https://github.com/rpush/rpush).

I had 28 migrations that pass the strong_migration check, and the rpush installation added 17 more that fail.  Is there any method like this that would allow me to skip a range of migrations?

Screen Shot 2021-01-18 at 11 45 02 AM

**Pull request closed without merging**